<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <!--works with multirobot_nv/launch/start_rosbot_only_map.launch-->
    <arg name="robot_namespace" default='rosbot1'/>
    <arg name="initial_pose_x" value="0.0"/> 
    <arg name="initial_pose_y" value="0.0"/> 
    <arg name="initial_pose_z" value="0.0"/>


    <arg name="odom_frame_id"   default="$(arg robot_namespace)/odom"/>
    <arg name="base_frame_id"   default="$(arg robot_namespace)/base_link"/>
    <arg name="global_frame_id" default="map"/>
    <arg name="odom_topic" default="$(arg robot_namespace)/odom" />
    <arg name="laser_topic" default="$(arg robot_namespace)/scan" />
    <!--add by ws-->
    <arg name="sensor_frame_id" default="$(arg robot_namespace)/laser" />
    <!--add by ws-->



    <group ns="$(arg robot_namespace)">
            <!--<param name="tf_prefix" value="$(arg robot_namespace)" />-->

        <node pkg="rplidar_ros" type="rplidarNode" name="rplidar">
            <param name="tf_prefix" value="$(arg robot_namespace)" />
            <param name="angle_compensate" type="bool" value="true"/>
            <param name="frame_id" type="string" value="$(arg robot_namespace)/laser"/>
            <param name="serial_baudrate" type="int" value="256000"/>
        </node>

        <!--replace rosbot_ekf/launch/all.launch-->
        <include file="$(find rosbot_ekf)/launch/rosserial_bridge.launch">
            <arg name="serial_port" value="/dev/ttyS4"/>
            <arg name="serial_baudrate" value="460800"/>
        </include>

        <include file="$(find rosbot_ekf)/launch/message_conversion.launch"/>

        <!--<include file="$(find rosbot_ekf)/launch/rosbot_ekf.launch"/>-->
        <node pkg="robot_localization" type="ekf_localization_node" name="rosbot_ekf" clear_params="true">
            <rosparam command="load" file="$(find rosbot_ekf)/params/ekf_params.yaml" />
            <param name="base_link_frame" value="$(arg robot_namespace)/base_link"/>
            <param name="odom_frame" value="$(arg robot_namespace)/odom"/>
            <param name="world_frame" value="$(arg robot_namespace)/odom"/>

            <!--  Placeholder for output topic remapping -->
            <remap from="odometry/filtered" to="$(arg robot_namespace)/odom"/>
            <!-- <remap from="accel/filtered" to=""/> -->
        </node>
        <node pkg="tf" type="static_transform_publisher" name="imu_publisher" args="0 0 0.02 0 0 0 $(arg robot_namespace)/base_link $(arg robot_namespace)/imu_link 100" />
        <!--replace rosbot_ekf/launch/all.launch-->

	<node pkg="tf" type="static_transform_publisher" name="laser_broadcaster" args="0 0 0 3.14 0 0 $(arg robot_namespace)/base_link $(arg robot_namespace)/laser 100" />
        

        <!--just for test-->
        <!--<node pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" name="teleop_twist_keyboard" output="screen"/>-->    
        <!--just for test-->

    </group>

</launch>
